File: Lesson_1/simple_linear_regression.R

# Importing the dataset
dataset = read.csv('Lesson_1/Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
library(caTools)
set.seed(123)
split <- sample.split(dataset$Salary, SplitRatio = 2/3) 
# This is a vector of boolean values. True means it goes to training set and False goes to test set.
training_set <- subset(dataset, split == TRUE)
test_set <- subset(dataset, split == FALSE)

# The model will learn the correlation between salary and yrs of experience in the training set and we will apply/ 
# use its power of prediction on the test set.

# Fitting Simple Linear Regression to the Training set
# This is where the coefficients are calculated. 
regressor <- lm(Salary ~ YearsExperience, data = training_set)
summary(regressor) # *** means it is highly statistically significant.

# Predicting the Test set results.
y_pred <- predict(regressor, newdata = test_set)

# Visualising the Training set results.
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'green') +
  ggtitle('Salary vs Experience (Training set)') +
  xlab('Years of experience') +
  ylab('Salary')

# Visualising the Test set results.
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'green') +
  ggtitle('Salary vs Experience (Test set)') +
  xlab('Years of experience') +
  ylab('Salary')


# Feature Scaling
# training_set[, 2:3] <- scale(training_set[, 2:3])
# test_set[, 2:3] <- scale(test_set[, 2:3])

File: Lesson_1/simple_linear_regression.py

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state = 0)

# Feature Scaling
'''from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)'''

# Fitting Simple Linear Regression to the Training Set.
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Needs fitting before predicting. Predicting the Test set results.
# Compare this with y_test which are the real values.
y_pred = regressor.predict(X_test)

# Visualising the Training set results.
plt.scatter(X_train, y_train, color = 'red')
# plot the predicted training test.
plt.plot(X_train, regressor.predict(X_train), color = 'blue') #regression/predicted line
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.sho()

# Next we need to plot the test sets against the blue (regression) line.
plt.scatter(X_test, y_test, color = 'red')
# plot the predicted training test.
plt.plot(X_train, regressor.predict(X_train), color = 'blue') #regression/predicted line
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.sho()


